src/ol_infrastructure/applications/bootcamps_ecommerce/__main__.py

```diff
@@ -0,0 +1,203 @@
+"""Create the infrastructure and services needed to support the bootcamps application.
+
+- Create a Redis instance in AWS Elasticache
+- Create a PostgreSQL database in AWS RDS for production environments
+- Create an IAM policy to grant access to S3 and other resources
+"""
+import json
+from pathlib import Path
+
+import pulumi_vault as vault
+from pulumi import Config, StackReference, export
+from pulumi_aws import ec2, iam, s3
+
+from bridge.lib.magic_numbers import DEFAULT_POSTGRES_PORT
+from bridge.secrets.sops import read_yaml_secrets
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLPostgresDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultPostgresDatabaseConfig,
+)
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+setup_vault_provider()
+bootcamps_config = Config("bootcamps")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+apps_vpc = network_stack.require_output("applications_vpc")
+data_vpc = network_stack.require_output("data_vpc")
+operations_vpc = network_stack.require_output("operations_vpc")
+aws_config = AWSBase(
+    tags={
+        "OU": "bootcamps",
+        "Environment": f"applications_{stack_info.env_suffix}",
+        "Application": "bootcamps",
+    }
+)
+
+# Create S3 buckets
+
+# Bucket used to store file uploads from bootcamps app.
+bootcamps_storage_bucket_name = f"ol-bootcamps-app-{stack_info.env_suffix}"
+bootcamps_storage_bucket = s3.Bucket(
+    f"ol-bootcamps-app-{stack_info.env_suffix}",
+    bucket=bootcamps_storage_bucket_name,
+    versioning=s3.BucketVersioningArgs(
+        enabled=True,
+    ),
+    policy=json.dumps(
+        {
+            "Version": "2008-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {"AWS": "*"},
+                    "Action": "s3:GetObject",
+                    "Resource": f"arn:aws:s3:::{bootcamps_storage_bucket_name}/courses/*",  # noqa: E501
+                }
+            ],
+        }
+    ),
+    tags=aws_config.tags,
+)
+
+
+bootcamps_iam_policy = iam.Policy(
+    f"bootcamps-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the bootcamps application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/bootcamps/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
+    policy=lint_iam_policy(
```

Comment:
Please pay careful attention to this IAM rule. The bucket already exists so will remain unchanged for now, but in a disaster recovery scenario, would these be the right perms with which to re-create?

Is there more testing I can do to validate my assertions here? I mostly copied/pasted from OCW at @blarghmatey's suggestion.